chapters/synchronous.tex

```diff
@@ -215,6 +215,16 @@ \subsection{Base-Clock and Sub-Clock Partitions}\label{base-clock-and-sub-clock-
 partition, i.e., synchronized when the ticks of the respective clocks are simultaneous.
 \end{definition}
 
+The terminology for the partitions are as follows:
+\begin{itemize}
+\item \willintroduce{partition|clocked}[Clocked (base-clock) partitions].
+\begin{itemize}
+\item \willintroduce{partition|discrete-time}[Discrete-time (sub-clock) partitions].
+\item \willintroduce{partition|discretized}[Discretized (sub-clock) partitions].
+\end{itemize}
+\item \willintroduce{partition|continuous-time}[Continuous (base-clock) partition].
```

Comment:
This is not the same as the continuous time partition that is not associated with any clocks, right?

My understanding (and it may be wrong) is that the partitions go as follows.

All equations and variables:
- Equations and variables not associated with any clock -> continuous-time partition (according to 16.7 and 16.7.3)
- Clocked equations and variables
  - Partitioned into which clock they are associated with -> base clock partitions
    - Discrete-time
    - Discretized
    - Subclock partitions
      - Discrete-time
      - Discretized
      
In case of the clocked partition both discrete and discretized are in terms of clocked variables. Any variable from the unclocked partition is treated as constant given by ``hold``. So, anything we are solving for would only be for clocked variables. And what determines whether it is discrete or discretized is the type of equations. Is that correct?

Maybe we should call equations and variables that are not associated with any clocks as "unclocked". Just to eliminate one use of "continuous". Also we call it as continuous-time partition in 16.7.3 but in 16.8.1 it says:
> If not explicitly stated otherwise, an expression with a variability such as continuous-time or discrete-time means that the expression is inside a partition that is not associated to a clock.
So it's not only continuous-time.

Reply:
> This is not the same as the continuous time partition that is not associated with any clocks, right?

It should be the normal continuous time partition.

> My understanding (and it may be wrong) is that the partitions go as follows.
> 
> All equations and variables:
> 
> * Equations and variables not associated with any clock -> continuous-time partition (according to 16.7 and 16.7.3)
> * Clocked equations and variables
>   
>   * Partitioned into which clock they are associated with -> base clock partitions
>     
>     * Discrete-time
>     * Discretized
>     * Subclock partitions
>       
>       * Discrete-time
>       * Discretized

I also thought something like that - but then I realized it is "simpler". The idea is instead that base-clock partitions are just "clocked" whereas sub-clock partitions can be discrete-time or discretized.

That allows a clocked base-clock partition to have a mix of discretized and discrete-time sub-clock partitions (even with different solverMethods).  That's why solverMethod can be inferred between sub-clock partitions (see Inferencing of solverMethod).

Unfortunately the specification doesn't make that straightforward, since the way to associate a solverMethod with a clock seems to associate it with a base-clock as well.

> In case of the clocked partition both discrete and discretized are in terms of clocked variables. Any variable from the unclocked partition is treated as constant given by `hold`. So, anything we are solving for would only be for clocked variables. And what determines whether it is discrete or discretized is the type of equations. Is that correct?
> 
> Maybe we should call equations and variables that are not associated with any clocks as "unclocked". Just to eliminate one use of "continuous". 

Hm... I guess that would make sense, and avoid the use of continuous-time. Will investigate after my vacation (if others agree) - possibly in a different PR.

> Also we call it as continuous-time partition in 16.7.3 but in 16.8.1 it says:
>
> > If not explicitly stated otherwise, an expression with a variability such as continuous-time or discrete-time means that the expression is inside a partition that is not associated to a clock.
> > So it's not only continuous-time.

Yes, "unclocked" would be better.

Reply:
> > Also we call it as continuous-time partition in 16.7.3 but in 16.8.1 it says:
> > > If not explicitly stated otherwise, an expression with a variability such as continuous-time or discrete-time means that the expression is inside a partition that is not associated to a clock.
> > > So it's not only continuous-time.
> 
> Yes, "unclocked" would be better.

Agreed.  I was trying to think of alternative names that would sound less exceptional and more like a default, considering that we're talking about what I guess will remain the main mode of operation in Modelica.  I came to think of _clock-free partition_ and _asynchronous partition_, but I admit that both of these share some of the original problem with _unclocked_.

Reply:
However, that is a separate PR #3399